### PR TITLE
Chore/upgrade lerna

### PR DESCRIPTION
**Pull request by dopenguin**
## Summary

Use `nx` with `npm workspaces` instead of `lerna`.

## Instructions for local reproduction and review

- `npm i`
- e.g. `npm run meldemichel:dev`
- e.g. `npm run meldemichel:build` 

## TODO

- [x] ~~By using `npm workspaces` (which is the recommended way for `nx`), the `makeDocs` script is no longer usable regarding the documentation of used plugins of a client as they are no longer being placed in the node_modules of a client.~~ Now uses the versions currently available in the repository. Might get confusing if we release major versions.

## Relevant tickets, issues, et cetera

#149 has to be merged before this

### CODE REVIEW

**Comment by dopenguin** [comment]
@warm-coolguy as mentioned in the PR description, the `makeDocs` script uses the versions of the packages located in the repository.
Do you deem that an issue and should we address that regarding differing versions of the clients? Or should simply we always try to keep the clients in sync with the current versions we are developing (which I'm favouring)

**Review by warm-coolguy — CHANGES_REQUESTED** [review]
Running `npm run pages:build && npm run pages:build:serve` will provide no contents for `http://127.0.0.1:8080/docs/snowbox/client-snowbox.html`, and the folder `pages/docs/snowbox` is actually empty.

🏓 @dopenguin

File: package.json [hunk 1]
```diff
@@ -51,7 +57,7 @@
     "@jest/types": "^29.3.1",
     "@types/geojson": "^7946.0.8",
     "@types/jest": "^29.2.5",
-    "@types/node": "^20.14.14",
+    "@types/node": "^20.14.8",
```
[the remark this answers is not in the document]
Comment (dopenguin) [reply package.json]: Good catch! Adjusted it 40e72e469398542bcd02621880a949b853a7fd2e

**Comment by dopenguin** [comment]
> Running `npm run pages:build && npm run pages:build:serve` will provide no contents for `http://127.0.0.1:8080/docs/snowbox/client-snowbox.html`, and the folder `pages/docs/snowbox` is actually empty.
> 
> > @warm-coolguy as mentioned in the PR description, the makeDocs script uses the versions of the packages located in the repository.
> > Do you deem that an issue and should we address that regarding differing versions of the clients? Or should simply we always try to keep the clients in sync with the current versions we are developing (which I'm favouring)
> 
> That is an issue as soon as we have a client where `^` doesn't suffice to update to the newest version, and since some breaking change releases are planned, that is pretty much imminent. Where does `nx` put those 'old' dependencies, right in the clients? In that case, the solution could be a rather small piece of fallback logic.
> 
> 🏓 @dopenguin

I've fixed the issue you've encountered; `snowbox` was the only client having `@polar`-dependencies as `dependencies` instead of `devDependencies`.

`nx` doesn't handle that, `npm workspaces` does that. I suppose those dependencies would be put in the `node_modules`-folder of the client itself, like you expect.
Do you see that as a change for this PR or something we can take a look at once we release the new major versions?

🏓 @warm-coolguy

**Review by warm-coolguy — APPROVED** [review]
> `nx` doesn't handle that, `npm workspaces` does that. I suppose those dependencies would be put in the `node_modules`-folder of the client itself, like you expect. Do you see that as a change for this PR or something we can take a look at once we release the new major versions?
> 
> 🏓 @warm-coolguy

Sure, we can delegate that to future us. But we'll have to be a little clever about it, or else we'll break the repository right on `main`. I guess there's some local publishing to do. 🤔 We'll see.

🚀 @dopenguin

**Pull request merged**